pythonx/UltiSnips/vim_state.py

```diff
@@ -126,6 +126,11 @@ def conserve(self):
         ec = byte2col(el, ebyte - 1)
         self._mode = _vim.eval('visualmode()')
 
+        # fix bug when vim option 'selection' is 'exclusive'.
+        if _vim.eval('&selection') == 'exclusive':
+            if not (sl == el and sbyte == ebyte):
```

Comment:
don't move cursor when you press `v` into visual mode .
vim will select the char under cursor, but position of mark `'<` is equal `'>` . 
the effect is the same as `inclusive` mode. do not need to be reduced by 1.